### ISSUE TITLE
Can't run JS function prompts when using node package

### ISSUE THREAD

**Issue by jameshfisher** [promptfoo/promptfoo#313]
I'm following [Using the node package](https://www.promptfoo.dev/docs/usage/node-package/) and [Input and output files](https://www.promptfoo.dev/docs/configuration/parameters), but I can't figure out how to get promptfoo to run my JS function.

My `test1.js`:

```
module.exports = async function ({ topic }) {
  return [
    {
      role: "system",
      content: `You're an angry pirate. Be concise and stay in character.`,
    },
    {
      role: "user",
      content: `Tell me about ${topic}`,
    },
  ];
};
```

My `eval.js`:

```
import promptfoo from "promptfoo";

const results = await promptfoo.evaluate(
  {
    prompts: ["file://./test1.js"],
    providers: ["openai:gpt-3.5-turbo"],
    tests: [
      {
        vars: {
          topic: "Hello world",
        },
      },
      {
        vars: {
          topic: "I'm hungry",
        },
      },
    ],
  },
  {
    maxConcurrency: 2,
  }
);

console.log(JSON.stringify(results, null, 2));
```

The output:

```
{
  "version": 2,
  "results": [
    {
      "provider": {
        "id": "openai:gpt-3.5-turbo"
      },
      "prompt": {
        "raw": "file://./test1.js",
        "display": "file://./test1.js"
      },
      "vars": {
        "topic": "I'm hungry"
      },
      "response": {
        "output": "I'm sorry, but I am unable to access or open local files. Is there anything specific you would like to know or discuss about the file \"test1.js\"?",
...
```

Of course, what I actually want to do is put the prompt function directly in the call, like:

```
const results = await promptfoo.evaluate(
  {
    prompts: [
      async function ({ topic }) {
        return [
          {
            role: "system",
            content: `You're an angry pirate. Be concise and stay in character.`,
          },
          {
            role: "user",
            content: `Tell me about ${topic}`,
          },
        ];
      }
    ],
    providers: ["openai:gpt-3.5-turbo"],
```

But this gives me a type error, plus a runtime error:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
```

How can I use a JS function as the prompt when using the node package?

**Comment by typpo** [comment]
Hi @jameshfisher,

> Of course, what I actually want to do is put the prompt function directly in the call

Prompt functions are actually not supported via the [node package](https://promptfoo.dev/docs/usage/node-package) yet.  #315 adds this functionality!  It will be published in the next release.

**Comment by jameshfisher** [comment]
Cool, thanks! ❤️

**Comment by typpo** [comment]
This fix is included in 0.31.0
